### PR TITLE
pretty-name: new recipe

### PR DESCRIPTION
**Pull request by Zeeno-atl**
Specify library name and version: **pretty-name/1.0.0**

Hello, I am the author of the "library" and I want to add it, since my other libraries that I want to package as well depend on this. I read all the guidelines (and went through some recipes as well), but it is my first contribution, so please add comments also about best practices, so I can make quality contributions in the future.

Not all PEP8 rules are satisfied, most notably missing docstrings and creating attributes outside of __init__ (e.g self.cmake = CMake(self) in build). But reading from other recipes, I have figured out that not all rules are enforced.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — DISMISSED** [review]

File: recipes/pretty-name/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 3.18)
+project(test_package LANGUAGES CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+add_executable(${PROJECT_NAME} src/test.cpp)
+conan_target_link_libraries(${PROJECT_NAME})
```
Comment (uilianries) [inline recipes/pretty-name/all/test_package/CMakeLists.txt:8]:
```suggestion

find_package(pretty-name CONFIG REQUIRED)

add_executable(${PROJECT_NAME} src/test.cpp)
target_link_libraries(${PROJECT_NAME} PRIVATE pretty-name::pretty-name)
```

File: recipes/pretty-name/all/test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    settings = "os", "compiler", "arch", "build_type"
+    generators = "cmake", "cmake_find_package_multi"
+
+    def build(self):
+        self.cmake = CMake(self)
+        self.cmake.configure()
+        self.cmake.build()
+
+    def build_requirements(self):
+        self.build_requires("cmake/3.18.0")
```
Comment (uilianries) [inline recipes/pretty-name/all/test_package/conanfile.py:14]:
```suggestion
```

Not needed

File: recipes/pretty-name/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 3.18)
```
Comment (uilianries) [inline recipes/pretty-name/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.4)
```

File: recipes/pretty-name/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    def test(self):
+        if not tools.cross_building(self.settings):
```
Comment (uilianries) [inline recipes/pretty-name/all/test_package/conanfile.py:17]:
```suggestion
        if not tools.cross_building(self):
```

File: recipes/pretty-name/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses",
+                  src=self._source_subfolder)
+        self.copy(pattern="*", dst="include",
+                  src=os.path.join(self._source_subfolder, "include"))
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "14")
```
Comment (uilianries) [inline recipes/pretty-name/all/conanfile.py:31]:
```suggestion
    @property
    def _minimum_compilers_version(self):
        return {
            "Visual Studio": "16",
            "gcc": "5",
            "clang": "5",
            "apple-clang": "5.1",
        }

    def validate(self):
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, "14")

        minimum_version = self._minimum_compilers_version.get(
            str(self.settings.compiler), False
        )
        if not minimum_version:
            self.output.warn(
                "pretty-name requires C++14. Your compiler is unknown. Assuming it supports C++14."
            )
        elif tools.Version(self.settings.compiler.version) < minimum_version:
            raise ConanInvalidConfiguration(
                "pretty-name requires C++14, which your compiler does not support."
            )
```

File: recipes/pretty-name/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import os
+from conans import ConanFile, tools
```
Comment (uilianries) [inline recipes/pretty-name/all/conanfile.py:2]:
```suggestion
from conans import ConanFile, tools
from conans.errors import ConanInvalidConfiguration


required_conan_version = ">=1.33.0"
```

File: recipes/pretty-name/all/test_package/conanfile.py (hunk 2 above)
Comment (Zeeno-atl) [reply recipes/pretty-name/all/test_package/conanfile.py]: Fixed in 7207616

File: recipes/pretty-name/all/test_package/CMakeLists.txt (hunk 3 above)
Comment (Zeeno-atl) [reply recipes/pretty-name/all/test_package/CMakeLists.txt]: Fixed in 7207616

File: recipes/pretty-name/all/test_package/CMakeLists.txt (hunk 1 above)
Comment (Zeeno-atl) [reply recipes/pretty-name/all/test_package/CMakeLists.txt]: Fixed in 3286a7b

File: recipes/pretty-name/all/conanfile.py (hunk 6 above)
Comment (Zeeno-atl) [reply recipes/pretty-name/all/conanfile.py:2]: Fixed in bd13020

File: recipes/pretty-name/all/conanfile.py (hunk 5 above)
Comment (Zeeno-atl) [reply recipes/pretty-name/all/conanfile.py]: Fixed in bd13020

File: recipes/pretty-name/all/test_package/conanfile.py (hunk 4 above)
Comment (Zeeno-atl) [reply recipes/pretty-name/all/test_package/conanfile.py]: Fixed in 187cc52

**Review by uilianries — APPROVED** [review]
LGTM

**Review by madebr — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Pull request merged**